### PR TITLE
feat: add /clear-screen slash command

### PR DESCRIPTION
**Pull request by VitalyOstanin**
## Summary

Add `/clear-screen` slash command that clears the screen without resetting conversation context.

- All messages remain in the `messages` array (full API context preserved)
- Messages are hidden from ink rendering via UUID-based filter on `g97`
- Terminal scrollback cleared via ANSI escape sequences
- Ink repaint triggered via `globalThis.__tweakccForceRedraw`
- Recommended keybinding: `ctrl+k ctrl+l` (Chat context)

## How it works

Three-part patch:

1. **Expose ink's `forceRedraw()`** — patches the `app:redraw` callback registration site to assign `globalThis.__tweakccForceRedraw` pointing to the ink instance's `forceRedraw()` method
2. **Patch render filter `g97`** — injects a check at the top of the render visibility function: if a message's UUID prefix is in `globalThis.__tweakccHiddenUUIDs`, return `false` (hide from UI). This works for both user and assistant messages, including multi-content assistant messages whose UUIDs are derived via `m$` (first 24 chars stay the same)
3. **Register `/clear-screen` command** — a `type:"local"` slash command that collects all current message UUID prefixes into a `Set` on `globalThis.__tweakccHiddenUUIDs`, clears terminal scrollback, and triggers ink repaint

### Why UUID-based hiding instead of replacing messages

The initial approach replaced the messages array via `setMessages`, keeping only the last assistant message with `content: []` to preserve ctx%. This caused Claude to lose conversation context — it would respond with "This is the first message in our session" because `setMessages` is the sole source for API messages. The UUID-based approach leaves all messages intact in the array (preserving full API context) and only hides them from ink rendering.

## Test plan

- [x] `/clear-screen` clears visible messages while preserving full conversation context
- [x] After `/clear-screen`, Claude remembers prior conversation (no "first message in session")
- [x] `ctrl+k ctrl+l` keybinding triggers the command (Chat context)
- [x] Terminal scrollback is cleared
- [x] Ink repaints correctly after clear
- [x] Unit tests pass (12 tests covering both `writeClearScreen` and `patchRenderFilter`: success, already patched, pattern not found, context preservation, delimiter variants, different function/argument names)
- [x] Tested on Claude Code 2.1.87 and 2.1.88

## Summary by CodeRabbit

* **New Features**
  * Adds a clear-screen slash command that clears the terminal, updates hidden-message tracking, and triggers a screen redraw without affecting hidden messages.

* **Tests**
  * Adds tests covering command injection, render-filter patching, preservation of original redraw behavior, and edge cases (already-present command, missing patterns, delimiter variations, parameter-name differences).

* **Chores**
  * Registers the clear-screen patch in the patch application pipeline.

### CODE REVIEW

File: src/patches/clearScreen.test.ts [hunk 1]
```diff
@@ -0,0 +1,55 @@
+import { describe, expect, it } from 'vitest';
```
[the remark this answers is not in the document]
Comment (VitalyOstanin) [reply src/patches/clearScreen.test.ts:1]: @coderabbitai Fixed in 829f7b6 — but the vitest import stays. While `vitest.config.ts` has `globals: true`, the project `tsconfig.json` does not include `"types": ["vitest/globals"]`, so `tsc --noEmit` (run by pre-commit hook via `pnpm lint`) fails without explicit imports. All other test files in the repo use explicit imports for the same reason.

File: src/patches/clearScreen.ts [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+  const commandDef =
+    ',{type:"local",name:"clear-screen",' +
+    'description:"Clear screen without resetting conversation context",' +
+    'supportsNonInteractive:!1,' +
+    'load:()=>Promise.resolve().then(()=>({call:(H,$)=>{' +
+    '$.setMessages(m=>{' +
+    'for(let i=m.length-1;i>=0;i--)' +
+    'if(m[i]?.type==="assistant"&&m[i].message?.usage)' +
+    'return[{...m[i],message:{...m[i].message,content:[]}}];' +
+    'return[]});' +
```
[the remark this answers is not in the document]
Comment (VitalyOstanin) [reply src/patches/clearScreen.ts]: @coderabbitai Fixed in 829f7b6. `setMessages` now preserves the last assistant message as fallback when no message has `message.usage`, instead of returning `[]`. Added edge-case test.

**Comment by bl-ue** [comment]
@coderabbitai resume

**Comment by bl-ue** [comment]
Hi @VitalyOstanin!  I'm sorry for the long delay on reviewing this PR!  The code looks great, but I ran it on CC v2.1.158 and got an error:

```
patch: clearScreen: failed to find app:redraw callback
```

Would you mind please taking a look at that so that I can merge your PR?  Thank you!

**Comment by VitalyOstanin** [comment]
Hi @bl-ue, thanks for the review and for testing on 2.1.158!

The failure (`failed to find app:redraw callback`) came from `redrawPattern`: recent Claude Code moved the stdout force-redraw call out of a `useCallback` into a standalone top-level function, so the old pattern no longer matched.

I've rebased the branch onto current `main` and fixed it:
- `redrawPattern` now anchors on the standalone `function NAME(){MAP.get(process.stdout)?.forceRedraw()}` form.
- The slash-command array lookup needed no change — it's covered by the rewritten `slashCommands.ts` from #761, which the branch now sits on top of.

Verified end to end on 2.1.158 and 2.1.167: the patch applies, the binary launches, and `/clear-screen` is registered and works.
